apps/courses/src/components/Icon.tsx

```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import { Icon, IconProps } from '@chakra-ui/react';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+
+/*
+ * This component is to be used when you are passing a FontAwesome icon
+ * to the Chakra UI Icon component. In the use cases, you will notice one
+ * particularity: there are boxSize and size to each component (when needed).
+ * This is the case because all FontAwesome icons, by default, come with a
+ * fa-w-16 class that limits the width of the icon to 16px. The best use case,
+ * in my opinion, so we don't end up writing and maintaining a dictionary
+ * matching the list of sizes from FontAwesome to Chakra sizes is to allow the
+ * components to have both `boxSize` and `size` props. `boxSize` will rule the
+ * Chakra side of the Icon, setting the height, while `size` will correctly
+ * scale the icon. The Chakra size should always "win" if you map it right.
+ */
```

Comment:
This makes sense, but it feels like a HUGE hack. Let's try the following things:

1. Not using `as={FontAwesomeIcon}` and instead trying to load the icon directly onto the `<Icon>` component. Screw react-fontawesome! I don't know that we need it at all and it would allow us to use `boxSize` naturally. I think when you receive the `icon={faMyIcon}` prop, you'll need (on this custom component) to destructure that icon and use one of it's properties. If you `console.log(faMyIcon)` then you'll see it's basically an object with an svg as one of the properties. Maybe we can use this directly and completely get rid of `react-fontawesome` and its awful sizing completely. That's what we should have done all along...
2. If that doesn't work, find a way to override the CSS of `fa-w-16` and make it 100% of the container. Wrap the custom component in a `<Box>` with a `boxSize` and the remaining passed props go on the `<Icon>` inside. That should force the icon to adapt to the width of its parent. Only do this if you can't get the first method to work.

Reply:
I disagree that this is a hack.

`boxSize` is a prop for Chakra's Icon and `size` is a prop that is passed down to react-fontawesome. This is as designed and it's not eye-catching or the most understandable thing in the world, I agree.

I'd rather use the SVGs directly.

I'll spin something new up.